lib/ansible/executor/module_common.py

```diff
@@ -317,11 +318,12 @@ def debug(command, zipped_mod, json_params):
             # Note: temp_path isn't needed once we switch to zipimport
             invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
     finally:
-        try:
-            shutil.rmtree(temp_path)
-        except (NameError, OSError):
-            # tempdir creation probably failed
-            pass
+        if not KEEP_FILES:
+            try:
+                shutil.rmtree(temp_path)
```

Comment:
This temp dir is never exposed anywhere, so a user would have to go hunting for this.  I question how useful this makes anything.

I think instead of this being the feature, a new command could be added to `debug` to allow re-creating the zip file, but I'm not sure how useful that would be either.

Reply:
`/tmp` is pretty obvious. There is the option to specify `dir` as scripts's basedir when creating `temp_path`, would that be appropriate?

Reply:
It may be obvious to you, but if it is not printed during the execution, it's not obvious, nor is simply adding a `dir` argument.  If it cannot be printed, which it cannot during the process of module execution, we should rely on something that is obvious.

Python can use any of the following by default:
'TMPDIR', 'TEMP', 'TMP', '/tmp', '/var/tmp', '/usr/tmp'

The user of Ansible, may not be responsible for the configuration of the system, and may be unfamiliar with where that zip would show up.  It's meant to be an implementation detail, and not to be relied on.

As mentioned, I'm not sure what use this actually provides, but I'd imagine it is just as useful to adding a command like `python AnsiballZ_ping.py create_zip` which would create the zip file in the tmp dir that is left behind when using `ANSIBLE_KEEP_REMOTE_FILES` and would be an explicit action.

In the current form, I cannot see us accepting this.  The above recommendation is the only option I would be inclined to entertain.